stock_cycle_count/models/stock_cycle_count_rule.py

```diff
@@ -0,0 +1,202 @@
+    _name = 'stock.cycle.count.rule'
+
+    @api.one
+    def _compute_currency(self):
+        self.currency_id = self.env.user.company_id.currency_id
+
+    @api.model
+    def _selection_rule_types(self):
+        return [
+            ('periodic', _('Periodic')),
+            ('turnover', _('Value Turnover')),
+            ('accuracy', _('Minimum Accuracy')),
+            ('zero', _('Zero Confirmation'))]
+
+    @api.one
+    @api.constrains('rule_type', 'warehouse_ids')
+    def _check_zero_rule(self):
+        if self.rule_type == 'zero' and len(self.warehouse_ids) > 1:
+            raise UserError(
+                _('Zero confirmation rules can only have one warehouse '
+                  'assigned.')
+            )
+        if self.rule_type == 'zero':
+            zero_rule = self.search([
+                ('rule_type', '=', 'zero'),
+                ('warehouse_ids', '=', self.warehouse_ids.id)])
+            if len(zero_rule) > 1:
+                raise UserError(
+                    _('You can only have one zero confirmation rule per '
+                      'warehouse.')
+                )
+
+    @api.onchange('rule_type')
+    def _get_rule_description(self):
+        if self.rule_type == 'periodic':
+            self.rule_description = _('Ensures that at least a defined number '
+                                      'of counts in a given period will '
+                                      'be run.')
+        elif self.rule_type == 'turnover':
+            self.rule_description = _('Schedules a count every time the total '
+                                      'turnover of a location exceeds the '
+                                      'threshold. This considers every '
+                                      'product going into/out of the location')
+        elif self.rule_type == 'accuracy':
+            self.rule_description = _('Schedules a count every time the '
+                                      'accuracy of a location goes under a '
+                                      'given threshold.')
+        elif self.rule_type == 'zero':
+            self.rule_description = _('Perform an Inventory Adjustment every '
+                                      'time a location in the warehouse runs '
+                                      'out of stock in order to confirm it is '
+                                      'truly empty.')
+        else:
+            self.rule_description = _('(No description provided.)')
+
+    @api.constrains('periodic_qty_per_period', 'periodic_count_period')
+    def _check_negative_periodic(self):
+        if self.periodic_qty_per_period < 1:
+            raise UserError(
+                _('You cannot define a negative or null number of counts per '
+                  'period.')
+            )
+        if self.periodic_count_period < 0:
+            raise UserError(
+                _('You cannot define a negative period.')
+            )
+
+    name = fields.Char('Name', required=True)
```

Comment:
That is what I was trying to say. The default label is working since v9.
Also there is a pylint-odoo to check this and I have configured my repos with travis to fail if you put the same name of the field.